data/src/main/java/io/gnosis/data/backend/dto/TransactionDto.kt

```diff
@@ -1,16 +1,60 @@
 package io.gnosis.data.backend.dto
 
+import pm.gnosis.model.Solidity
+
 data class DataDecodedDto(
     val method: String,
-    val parameters: List<ParamsDto>?
+    val parameters: List<ParamDto>?
 )
 
-data class ParamsDto(
-    val name: String,
-    val type: String,
-    val value: Any
+data class ValueDecodedDto(
+    val operation: Operation,
+    val to: Solidity.Address,
+    val value: Long,
+    val data: String,
+    val dataDecoded: DataDecodedDto?
 )
 
+sealed class ParamDto {
+    abstract val type: String
+    abstract val name: String
+    abstract val value: Any?
+
+    data class AddressParam(
+        override val type: String,
+        override val name: String,
+        override val value: Solidity.Address
+    ) : ParamDto()
+
+    data class ArrayParam(
+        override val type: String,
+        override val name: String,
+        override val value: List<Any>
```

Comment:
Does this have to be `Any` ? I think that for the sake of regularity it would make more sense to leave it as a `String`

Reply:
Would also be easier to put into the UI

Reply:
It can be list of list of list ....

Reply:
What we did in the gateway was to have something like a sealed class called `ParamValue` and 2 subclasses `SingleValue` and `ArrayValue`. Then you could just have `val value: ParamValue` and the type gets resolve upon deserialization.

Reply:
Unfortunately, moshi needs to know the concrete type or have a concrete polymorphic adapter registered. We also cannot register polymorphic adapters for all solidity type combinations